src/main/java/course_project/sea_battle/service/inputs/InputReader.java

```diff
@@ -0,0 +1,15 @@
+package course_project.sea_battle.service.inputs;
+
+import java.util.Scanner;
+
+public abstract class InputReader {
+    protected Scanner scanner;
+
+    public String readLine() {
+        return scanner.nextLine();
+    }
+
+    public Scanner getScanner() {
+        return scanner;
+    }
+}
```

Comment:
The purpose of this class in also inclear

Reply:
This is the only question I have: how could I avoid this abstraction and use tests in this way:
String input = "a1 h\n" +
                "a3 h\n" +
                ...
                "\n";
        setInput(input);
Because if I have Scanner.in inside class I am testing I have the same exception all the time: no line found. So I solved it this way. Solution for tests...